README.md

```diff
@@ -40,24 +46,131 @@ If you intend to use the software constantly, you can install for the user or at
+
+```bash
+$ sudo apk add libstdc++
+```
+
+Download the *h1-alpine* file from [Releases] using wget:
+
+```bash
+$ wget https://github.com/hyperonecom/h1-cli/releases/latest/download/h1-alpine
+```
+
+Then run the downloaded binary file after granting executables rights:
+
+```bash
+$ chmod +x h1-alpine
+$ ./h1-alpine
+```
+
+The application should work correctly.
+
+If you intend to use the software constantly, you can install for the user or at the system level.
+
+Installation for the user can be carried out by:
+
+```bash
+$ mkdir -p ~/.local/bin/
+$ export PATH=${PATH}:~/.local/bin/ >> ~/.bash_profile
+$ source ~/.bash_profile
+$ cp ./h1-alpine ~/.local/bin/h1
+$ h1
+```
+
+Installation at the system level can be carried out by:
+
+```bash
+$ sudo cp h1-alpine /usr/local/bin/h1
 $ h1
 ```
 
 ### Windows environment
 
-> Introduction for Windows requires development taking into account the specificity of the platform.
+Download the *h1-win.exe* file from [Releases] using Invoke-WebRequest in Powershell:
+
+```bash
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+Invoke-WebRequest https://github.com/hyperonecom/h1-cli/releases/latest/download/h1-win.exe -OutFile h1.exe
+```
+
+Then run the downloaded executable file using Powershell:
+
+```bash
+./h1
+```
+
+The application should work correctly without installing additional libraries / packages.
+
+If you intend to use the software constantly, you can add CLI to environment variables for the user or at the system level.
+
+Adding CLI to environment variables for the user can be carried out in Powershell by:
+
+```bash
+New-Item -ItemType directory -Path $ENV:UserProfile\h1\
+Copy-Item .\h1.exe $ENV:UserProfile\h1\
+[Environment]::SetEnvironmentVariable("Path", [Environment]::GetEnvironmentVariable("Path", "User") + ";$ENV:UserProfile\h1\", "User")
+h1
+```
+
+Adding CLI to environment variables at the system level can be carried out in Powershell by:
+
+```bash
+New-Item -ItemType directory -Path $ENV:Public\h1\
+Copy-Item .\h1.exe $ENV:Public\h1\
+[Environment]::SetEnvironmentVariable("Path", [Environment]::GetEnvironmentVariable("Path", "Machine") + ";$ENV:Public\h1\", "Machine")
+h1
+```
 
 ### macOS environment
 
-> Introduction for the macOS system requires development taking into account the specificity of the platform.
+Download the *h1-macos* file from [Releases] using wget:
+
+```bash
+$ wget https://github.com/hyperonecom/h1-cli/releases/latest/download/h1-macos
+```
+
+Then run the downloaded binary file after granting executables rights:
+
+```bash
+$ chmod +x h1-macos
+$ ./h1-macos
+```
+
+The application should work correctly without installing additional libraries / packages.
+
+If you intend to use the software constantly, you can install for the user or at the system level.
+
+Installation for the user can be carried out by:
+
+```bash
+$ mkdir ~/bin
+$ echo export "PATH=${PATH}:~/bin" >> ~/.bash_profile
```

Comment:
It is standard path on MacOS for this kind of stuffs?

Reply:
My first impression is that it should to go /usr/local/bin on MacOs. Something to discover for @yrself

Reply:
Take in account "user installation" and "system installation". For "user installation" we should not put it out of home directory.

Reply:
@gregorybrzeski just as @ad-m wrote, this is the case of "user installation". @ad-m this is not standard path because "~/bin" doesn't exist by default but various answers on Superuser or Stackoverflow used it.

Reply:
Since when Superuser and Stackoverflow sets standards for OS ? Does MacOS have a standard way to do "user installations" ? Does MacOS differentiate between user and system installation at all ?

Reply:
No, MacOS doesn't have a standard way of "user installations" and doesn't differentiate between these installations.

Reply:
So where MacOS expects users to add binaries/apps for terminal only use ?

Reply:
Either in /usr/local/bin or /usr/bin - these are by default in PATH

Reply:
Can you link to source of that information ?

Reply:
[Apple page for file system basics](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html)
> /usr—Contains non-essential command-line binaries, libraries, header files, and other data.

Reply:
What about /usr/local/bin and /usr/bin which you did mentioned earlier ?

Reply:
@gregorybrzeski see few lines below for "installation at the system level".

Reply:
@ad-m  are you sure we are looking at the document linked by @yrself ? I am unable to find quoted phrase in this document :(

We should allow @yrself to do discovery in this area and gather all relevant information and present it in concise way as an argument for the decision where files should be placed on MacOS.

Reply:
From previously linked Apple page: "The directories in this category are inherited from traditional UNIX installations" - so we should decide according to Filesystem Hierarchy Standard which [says](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html):

> The /usr/local hierarchy is for use by the system administrator when installing software locally.
> Locally installed software must be placed within /usr/local rather than /usr unless it is being installed to replace or upgrade software in /usr.

So I think now that there should be only information about installing the file in /usr/local/bin directory ("installation at the system level") and leave out information about "user installation"